pkg/filter/requestadaptor/requestadaptor.go

```diff
@@ -151,6 +159,31 @@ func (ra *RequestAdaptor) handle(ctx context.HTTPContext) string {
 			ctx.Request().SetHost(ra.spec.Host)
 		}
 	}
+
+	if ra.spec.Decompress != "" {
+		res := ra.processDecompress(ctx)
+		if res != "" {
+			return res
+		}
+	}
```

Comment:
there are 3 cases now:
* when `ra.spec.Body` is set, no need to do decompress
* when `ra.spec.Body` is not set, and decompress is set, decompress the body
* when compress (need to add) is set, compress the body
note that decompress and compress cannot be set at the same time.

Reply:
i agree with last two cases, but for first case, it may not happen... if set body, the header of "Content-Encoding" is empty, then it will not decompress anyway. if set the body but original body is compressed, then user should also set `Header.Del` to delete "Content-Encoding" because the new body is not compressed. if set body, do we need to delete header of "Content-Encoding" automatically?

Reply:
`if set body, the header of "Content-Encoding" is empty`, this is not true when the original request is gzipped.
`if set body, do we need to delete header of "Content-Encoding" automatically?`: yes.

Reply:
updated!